amx/server/events.lua

```diff
@@ -29,19 +34,25 @@ function gameModeInit(player)
 	g_Players[playerID].streamedVehicles = {}
 	g_Players[playerID].streamedPlayers = {}
 	g_Players[playerID].attachedObjects = {}
+	g_Players[playerID].shotVect = {
+		oX = 0.0, oY = 0.0, oZ = 0.0,
+		hX = 0.0, hY = 0.0, hZ = 0.0
+	}
+	g_Players[playerID].conntick = getTickCount()
 	g_Players[playerID].viewingintro = true

```

Comment:
Let's also initialize the state as `PLAYER_STATE_NONE`.

```suggestion
	g_Players[playerID].viewingintro = true
        g_Players[playerID].state = PLAYER_STATE_NONE
```